src/aspire/operators/wemd.py

```diff
@@ -10,52 +10,69 @@
     CAR-TR-1025 CS-TR-4908 UMIACS-TR-2008-06.
 """
 
+import warnings
+
 import numpy as np
 import pywt
 
 
-def wemd_embed(arr, wavelet, level):
+def wemd_embed(arr, wavelet="coif3", level=None):
     """
     This function computes an embedding of Numpy arrays such that
     for non-negative arrays that sum to one, the L1 distance between the
     resulting embeddings is strongly equivalent to the Earthmover distance
     of the arrays.
 
     :param arr: Numpy array
-    :param level: Decomposition level of the wavelets
-    Larger levels yield more coefficients and more accurate results
+    :param level: Decomposition level of the wavelets.
+    Larger levels yield more coefficients and more accurate results.
+    If no level is given, we take the the log2 of the side-length of the domain.
     :param wavelet: Either the name of a wavelet supported by PyWavelets
-    (e.g. 'coif3', 'sym3') or a pywt.Wavelet object
+    (e.g. 'coif3', 'sym3', 'sym5', etc.) or a pywt.Wavelet object
     See https://pywavelets.readthedocs.io/en/latest/ref/wavelets.html#built-in-wavelets-wavelist
+    The default is 'coif3', because it seems to work well empirically.
     :returns: One-dimensional numpy array containing weighted details coefficients.
     """
+    dimension = arr.ndim
 
-    arrdwt = pywt.wavedecn(arr, wavelet, mode="zero", level=level)
+    if level is None:
+        level = int(np.ceil(np.log2(max(arr.shape)))) + 1
 
-    dimension = len(arr.shape)
+    # Using wavedecn with the default level creates this boundary effects warning.
+    # However, this doesn't seem to be a cause for concern.
+    warnings.filterwarnings(
```

Comment:
What's the scope of this filter? Does it stick after we exit the method?

Reply:
Its not clear (and probably is not scoped to the function).  I'll wrap it in a context block. Give me a few minutes.

Reply:
should be better now. Good catch thanks.

Reply:
Looks like this is the way to do it
```
with warnings.catch_warnings():
    warnings.filterwarnings(…)
    do_stuff()
```
(see https://docs.python.org/3/library/warnings.html#temporarily-suppressing-warnings).

Reply:
Nm @garrettwrong beat me to it…